### PR TITLE
{Graph} az aro module migration to Microsoft Graph

### PR DESCRIPTION
**Pull request by nontw**
**Related command**
az aro

**Description**
Migrating from Azure SDK's AAD Graph to Microsoft Graph 
#22174 

**Testing Guide**
All the commands were executed successfully.

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [x] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [x] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by nontw** [comment]
@jiasli I was unable to re-record new set of live tests - randomly got 404 id not found error after the app was created.  Will ping you for more details.

**Comment by yonzhan** [comment]
Graph

File: src/azure-cli/azure/cli/command_modules/aro/_aad.py [hunk 1]
```diff
@@ -3,109 +3,55 @@
...
+            return sps[0]["id"]
         return None
 
     def get_application_by_client_id(self, client_id):
-        apps = list(self.client.applications.list(
-            filter="appId eq '%s'" % client_id))
+        apps = self.client.application_list(f"appId eq '{client_id}'")
         if apps:
-            return apps[0]
+            return apps[0]["id"]
```
Comment (jiasli) [inline src/azure-cli/azure/cli/command_modules/aro/_aad.py:39]: Perhaps the method should be renamed to `get_application_id_by_client_id` to better reflect what it is doing?

File: src/azure-cli/azure/cli/command_modules/aro/custom.py [hunk 2]
```diff
@@ -130,7 +129,7 @@ def aro_create(cmd,  # pylint: disable=too-many-locals
         ],
     )
 
-    sp_obj_ids = [client_sp.object_id, rp_client_sp.object_id]
+    sp_obj_ids = [client_sp, rp_client_sp]
```
Comment (jiasli) [inline src/azure-cli/azure/cli/command_modules/aro/custom.py:132]: Same here. Perhaps:

```suggestion
    sp_obj_ids = [client_sp_id, rp_client_sp_id]
```

File: src/azure-cli/azure/cli/command_modules/aro/_aad.py [hunk 3]
```diff
@@ -3,109 +3,55 @@
...
-                    custom_key_identifier=str(start_date).encode(),
-                    start_date=start_date,
-                    end_date=end_date,
-                    value=password,
-                ),
-            ],
-        ))
-
-        return app, password
+        request_body = {"displayname": display_name, }
```
Comment (jiasli) [inline src/azure-cli/azure/cli/command_modules/aro/_aad.py:23]: Please adhere to Graph API spec: https://docs.microsoft.com/en-us/graph/api/resources/application?view=graph-rest-1.0

```suggestion
        request_body = {"displayName": display_name}
```

File: src/azure-cli/azure/cli/command_modules/aro/_aad.py [hunk 4]
```diff
@@ -3,109 +3,55 @@
...
-or manually create a new client secret and run `az aro update --client-secret <ClientSecret>`.")
-
-        # when appending credentials ALL fields must be present, otherwise
-        # azure gives ambiguous errors about not being able to update old keys
-        credentials.append(PasswordCredential(
-            custom_key_identifier=str(start_date).encode(),  # bytearray
-            key_id=str(key_id),
-            start_date=start_date,
-            end_date=end_date,
-            value=password))
```
Comment (jiasli) [inline src/azure-cli/azure/cli/command_modules/aro/_aad.py:107]: Glad to know `addPassword` no longer requires all fields to be present: https://docs.microsoft.com/en-us/graph/api/application-addpassword?view=graph-rest-1.0&tabs=http

**Comment by jiasli** [comment]
@nontw, I have looped you in the internal email for this 404 issue.

The best way to solve it is to `time.sleep` for a while, like I am doing in `role` module:

https://github.com/Azure/azure-cli/blob/cac3f06cf483d8c81f93c9cd556116e27aff2182/src/azure-cli/azure/cli/command_modules/role/tests/latest/test_role.py#L397

You don't need to worry about its efficiency since it is mocked out during playback.

File: src/azure-cli/azure/cli/command_modules/aro/_aad.py (hunk 4 above)
Comment (nontw) [reply src/azure-cli/azure/cli/command_modules/aro/_aad.py:107]: Yeah, that's right.  I tested it and it worked without these parameters.  In fact, you can no longer set the password and custom key identifier.  The end date is also always less than 2 years.

File: src/azure-cli/azure/cli/command_modules/aro/_aad.py (hunk 1 above)
Comment (nontw) [reply src/azure-cli/azure/cli/command_modules/aro/_aad.py:39]: Pushed a new commit to incorporate this.

File: src/azure-cli/azure/cli/command_modules/aro/custom.py (hunk 2 above)
Comment (nontw) [reply src/azure-cli/azure/cli/command_modules/aro/custom.py]: Yes, this is incorporated in my new commit.

**Comment by nontw** [comment]
Add test recordings.

**Comment by yonzhan** [comment]
Please resolve the conflicting files.

**Comment by yonzhan** [comment]
Graph

**Review by jiasli — APPROVED** [review]

**Review by evelyn-ys — APPROVED** [review]

**Pull request merged**